### DISCUSSION TITLE
How can I guarantee exactly-once execution when updating key values in redis?

### DISCUSSION THREAD

**Discussion by kraspasov** [redis/lettuce#2767] [General]
I would like to know if there is a way to guarantee exactly-once execution for commands. The situation I am in is pretty specific:

1. We are using Redis to cache some application data during user interactions. During those interactions values might change, this is when we update the values.
2. For some reason the connections gets dropped occasionally on write acknowledgement. This is when the command gets queued for re-execution, but before that an update due to user interactions is done. After the update the re-execution takes place, thus updating the value to the previous one.

What can I do to guarantee only the most recent value is the one persisted in Redis?

**Comment by tishun** [comment]
Hey @kraspasov ,

There are three settings for commands reliability: **at-most-once**, **at-lease-once** and **exactly-once**, as per the [reference guide](https://lettuce.io/core/release/reference/#command-execution-reliability.what-does-at-most-once-mean).

If you are willing to accept the occasional lost commands the **at-most-once** setting seems to fit your needs:
> **at-most-once** execution means that for each command handed to the mechanism, that command is execution zero or one time; in more casual terms it means that commands may be lost.

I assume this should not be a huge problem if you are using Redis as a cache mechanism?
